spec/system/group_category_controllers_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+RSpec.describe 'GroupCategories', type: :system do
+  include Devise::Test::IntegrationHelpers
+  before(:each) do
+    @user = User.create(
+      name: 'Juwairiyya',
+      email: 'juwairiyyasadiq@gmail.com',
+      password: 'password'
+    )
+    @category = GroupCategory.create(
+      author_id: @user.id,
+      name: 'Juwairiyya',
+      icon: 'icon'
+    )
+    sign_in @user
+  end
+  describe 'GroupCategory Index' do
+    before(:each) do
+      visit group_categories_path
+    end
+    it 'should display the category name' do
+      expect(page).to have_content(@category.name)
+    end
+    it 'should display the category icon' do
+      expect(page).to have_css('i')
+    end
+    it 'should display the total amount for each category' do
+      expect(page).to have_content(@category.total_amount)
+    end
```

Comment:
- [ ] There are 4 tests failing, Please make sure that all tests are passing.

![image](https://github.com/Zuwairanajma/Budget_App/assets/68796232/91e02619-830f-4986-b191-df48523ded8a)